### DISCUSSION TITLE
Support for SBARRO experiments

### DISCUSSION THREAD

**Discussion by jamesnemesh** [broadinstitute/Drop-seq#266] [General]
### Support for SBARRO experiments 
- TagReadWithRabiesBarcodes - Tags an unaligned BAM with rabies virus barcode sequences and associated metrics. Barcode sequences are extracted using invariant sequences which flank the barcode regions. 
- FilterValidRabiesBarcodes - Filters rabies virus tags into pass/fail BAMs based on metrics associated with the extracted viral barcodes. 
- BipartiteRabiesVirusCollapse - Collapse rabies virus sequences in which half the barcode matches within some edit distance, within a cell.
 
### Other updates 
- TrimStartingSequence update:  New algorithm with mismatch rate and sequence not require to be anchored to start of read.  Note that different sequence is recommended for 10X and Drop-seq.  Old, buggy behavior can be enabled with LEGACY=true.
- PolyATrimmer now optionally stores length of polyA and trimmed adapter
- ConvertTagToReadGroup improved error messages when read group missing from input data
- FilterBamByGeneFunction - new program to filter reads to specified gene functions, in the same way DigtialGeneExpression and related programs internally filter data.
- CensusSeq and related software can now take as input a list of BAMs. 
- Csi analysis now generates a contamination confidence interval.
- Improvements to make GTF parsing more robust

<hr /><em>This discussion was created from the release <a href='https://github.com/broadinstitute/Drop-seq/releases/tag/v2.4.1'>Support for SBARRO experiments</a>.</em>

**Comment by tlz4320** [comment]
Hello! I'm reading SBARRO paper these days. But I didn't find the code to calculate whether the VBC is unique in this code repo. Could you please provide how to find the unique VBC. This is a very important part algorithm in SBARRO paper.
